src/ILLink.RoslynAnalyzer/TrimAnalysis/HandleCallAction.cs

```diff
@@ -41,9 +46,39 @@ public HandleCallAction (in DiagnosticContext diagnosticContext, ISymbol owningS
 		private partial MethodThisParameterValue GetMethodThisParameterValue (MethodProxy method, DynamicallyAccessedMemberTypes dynamicallyAccessedMemberTypes)
 			=> new (method.Method, dynamicallyAccessedMemberTypes);
 
+		private partial MethodParameterValue GetMethodParameterValue (MethodProxy method, int parameterIndex, DynamicallyAccessedMemberTypes dynamicallyAccessedMemberTypes)
+			=> new (method.Method.Parameters[parameterIndex], dynamicallyAccessedMemberTypes);
+
+		private partial IEnumerable<SystemReflectionMethodBaseValue> GetMethodsOnTypeHierarchy (TypeProxy type, string name, BindingFlags? bindingFlags)
+		{
+			foreach (var method in type.Type.GetMethodsOnTypeHierarchy (m => m.Name == name, bindingFlags))
+				yield return new SystemReflectionMethodBaseValue (new MethodProxy (method));
```

Comment:
I think this is going to cause a quadratic number of calls in the depth of the tree. Since iterators are lazy, we'll walk all the way down the tree to fetch the first element, then walk n-1 levels to fetch the second, and so on.

Reply:
Sorry I don't understand. The GetMethodsOnTypeHierarchy helper we're calling here already uses iterator, so this is basically just a pass through...

I guess I could rewrite both to use ImmutableArray...

I must admit I don't understand what the problem is.